In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

# Use the kagglehub client library to attach Kaggle resources like competitions, datasets, and models to your session
# Learn more about kagglehub: https://github.com/Kaggle/kagglehub/blob/main/README.md

import kagglehub
# kagglehub.dataset_download('<owner>/<dataset-slug>')

/kaggle/input/competitions/titanic/train.csv
/kaggle/input/competitions/titanic/test.csv
/kaggle/input/competitions/titanic/gender_submission.csv


# **Step 1: Import Libraries**

In [2]:
# Data Manipulation
import pandas as pd
import numpy as np

# Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# **Step 2: Load Dataset**

In [3]:
# Load Training and Test Dataset

train = pd.read_csv("/kaggle/input/competitions/titanic/train.csv")
test = pd.read_csv("/kaggle/input/competitions/titanic/test.csv")

# **Step 3: Dataset Overview**

In [4]:
# Shape of Dataset

print("Train Shape:", train.shape)
print("Test Shape:", test.shape)

Train Shape: (891, 12)
Test Shape: (418, 11)


# **Step 4: View First 5 Rows**

In [5]:
# Display First 5 Rows

train.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# **Step 5: Understand the Dataset Columns**

In [6]:
# Display Column Names

train.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

# **Step 6: Dataset Information**

In [7]:
# Display Dataset Information

train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


# **Step 7: Check Missing Values**

In [8]:
# Check Missing Values

train.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

# **Step 8: Check Duplicate Rows**

In [9]:
# Check Duplicate Rows

duplicate_rows = train.duplicated().sum()

print("Total Duplicate Rows:", duplicate_rows)

Total Duplicate Rows: 0


# **Step 9: Check Missing Values Percentage**

In [10]:
# Check Missing Values Percentage

missing_percentage = (train.isnull().sum() / len(train)) * 100

missing_percentage.sort_values(ascending=False)

Cabin          77.104377
Age            19.865320
Embarked        0.224467
PassengerId     0.000000
Name            0.000000
Pclass          0.000000
Survived        0.000000
Sex             0.000000
Parch           0.000000
SibSp           0.000000
Fare            0.000000
Ticket          0.000000
dtype: float64

# **Step 10: Analyze Missing Values**

### Missing Values Analysis

- **Cabin:** 77.10% missing values
- **Age:** 19.87% missing values
- **Embarked:** 0.22% missing values

### Initial Observations

- Cabin has a very high percentage of missing values.
- Age has a moderate number of missing values.
- Embarked has only a few missing values.

# **Step 11: Drop Cabin Column**

### Why are we dropping the Cabin column?

The **Cabin** column contains **77.10% missing values** (687 out of 891 records).

Since most of the values are missing, filling them would introduce too much uncertainty.

Therefore, we will remove this column from both the training and test datasets.

In [11]:
# Drop Cabin Column

train = train.drop(columns=["Cabin"])
test = test.drop(columns=["Cabin"])

print("Cabin column removed successfully.")

Cabin column removed successfully.


# **Step 12: Check Age Distribution**

In [12]:
# Summary Statistics for Age

train["Age"].describe()

count    714.000000
mean      29.699118
std       14.526497
min        0.420000
25%       20.125000
50%       28.000000
75%       38.000000
max       80.000000
Name: Age, dtype: float64

# **Step 13: Fill Missing Values in Age**

### Why are we using the Median?

The **Age** column has **177 missing values** (19.87%).

Since the median is less affected by outliers than the mean, we will use the **median value** to fill the missing ages.

In [13]:
# Fill Missing Values in Age

train["Age"] = train["Age"].fillna(train["Age"].median())
test["Age"] = test["Age"].fillna(test["Age"].median())

print("Missing values in Age handled successfully.")

Missing values in Age handled successfully.


# **Step 14: Handle Missing Values in Embarked**

### Why are we using the Mode?

The **Embarked** column has only **2 missing values**.

Since it is a categorical feature, we will fill the missing values with the **most frequent (mode)** category.

In [14]:
# Fill Missing Values in Embarked

train["Embarked"] = train["Embarked"].fillna(train["Embarked"].mode()[0])

print("Missing values in Embarked handled successfully.")

Missing values in Embarked handled successfully.


# **Step 15: Verify Missing Values**

In [15]:
# Verify Missing Values After Data Cleaning

train.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

# **Step 16: Data Cleaning Summary**

## Step 16: Data Cleaning Summary

### Data Cleaning Completed Successfully

The following preprocessing steps were performed:

- Removed the **Cabin** column due to 77.10% missing values.
- Filled missing values in the **Age** column using the **median**.
- Filled missing values in the **Embarked** column using the **mode**.
- Verified that there are **no missing values** remaining in the training dataset.

The dataset is now ready for Exploratory Data Analysis (EDA).

# **Step 17: Target Variable Analysis**

In [16]:
# Check Survival Distribution

train["Survived"].value_counts()

Survived
0    549
1    342
Name: count, dtype: int64

# **Step 18: Visualize Target Variable**